frontend/src/components/TopBar/TopBar.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import {
+  AppBar,
+  Button,
+  Toolbar,
+  IconButton,
+  Typography,
+  Grid,
+  Avatar,
+  Hidden,
+} from '@material-ui/core';
+import { ArrowBackIcon, MenuIcon, productIcon } from '../../assets/index';
+import { GRAY_1, GRAY_5, GRAY_6, SUNFLOWER } from '../../constants/index';
+
+export default function TopBar({
+  darkMode,
+  showArrow,
+  onActionIconClick, // For arrow back or navigation drawer icons
+  title,
+  iconsToShow = [], // Array of Icon Components(like in DownloadIcon.js)
+  onIconsClick = [], // Array of on click callbacks for each icon in iconsToShow. NOTE: order of callbacks must match order of icons
```

Comment:
@AArlauskas @kibartas Not sure whether this was a right choice to use two arrays, maybe it would be better to use `dict` with `key` being the icon object and `value` being the callback function.